src/backup.cpp

```diff
@@ -39,6 +41,72 @@ std::string GetBackupFilename(const std::string& basename, const std::string& su
     return rpath.string();
 }
 
+bool BackupsEnabled()
+{
+    // If either of these configuration options is explicitly set to zero,
+    // disable backups completely:
+    return GetArg("-walletbackupinterval", 1) > 0
+        && GetArg("-walletbackupintervalsecs", 1) > 0;
+}
+
+int64_t GetBackupInterval()
+{
+    int64_t backup_interval_secs = GetArg("-walletbackupintervalsecs", 86400);
+
+    // The deprecated -walletbackupinterval option specifies the backup interval
+    // as the number of blocks that pass. If someone still uses this in a config
+    // file, we'll honor it for now:
+    //
+    if (mapArgs.count("-walletbackupinterval")) {
+        backup_interval_secs = GetArg("-walletbackupinterval", 900) * 90;
+    }
+
+    return backup_interval_secs;
+}
+
+void RunBackupJob()
+{
+    TRY_LOCK(cs_main, locked_main);
+
+    if (!locked_main) {
+        return;
+    }
+
+    TRY_LOCK(pwalletMain->cs_wallet, locked_wallet);
+
+    if (!locked_wallet) {
+        return;
+    }
+
+    const int64_t now = GetSystemTimeInSeconds();
+
+    static const int64_t interval = GetBackupInterval();
+    static int64_t last_backup_time = pwalletMain->GetLastBackupTime();
+
+    // The scheduler runs this job at a faster rate than the configured backup
+    // interval in case this function skips a cycle because of lock contention
+    // in a busy wallet, so we double check the time:
+    //
+    if (now < last_backup_time + interval) {
+        return;
+    }
+
+    // Store the last backup time to the wallet file so that we can resume the
+    // configured backup schedule between node restarts:
+    //
+    pwalletMain->StoreLastBackupTime(now);
+
+    const bool wallet_result = BackupWallet(*pwalletMain, GetBackupFilename("wallet.dat"));
+    const bool config_result = BackupConfigFile(GetBackupFilename("gridcoinresearch.conf"));
+
```

Comment:
We need to run maintain backups here too. I will submit a PR against your branch...

Reply:
Looks good... let me know when you're finished and I'll move out of draft.